platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
     case fullAccuracy
 }
 
-class Coordinator: NSObject, MLNMapViewDelegate {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
-    var pannedToUserLocation = false
-
-    init(mapView: Binding<MLNMapView>, locationAccuracy: Binding<LocationAccuracyState>) {
-        _mapView = mapView
-        _locationAccuracy = locationAccuracy
+class MapViewModel: NSObject, ObservableObject {
+    // Weak because the coordinator has a strong reference to the view model
+    weak var mapCoordinater: MapLibreRepresentableCoordinator?
+    @MainActor @Published var locationAccuracy: LocationAccuracyState = .unknown
+
+    @MainActor func requestTemporaryLocationAuthorization() {
+        print("Requesting precice location")
+
+        switch locationAccuracy {
+        case .reducedAccuracy:
+            let purposeKey = "MLNAccuracyAuthorizationDescription"
+            mapCoordinater?.mapView?.locationManager.requestTemporaryFullAccuracyAuthorization!(withPurposeKey: purposeKey)
+        default:
+            break
+        }
     }
+}
 
-    func mapView(_: MLNMapView, didChangeLocationManagerAuthorization manager: MLNLocationManager) {
+class MapLibreRepresentableCoordinator: NSObject, MLNMapViewDelegate {
+    private var mapViewModel: MapViewModel
+    // Weak reference because SwiftUI owns the strong reference
+    private(set) weak var mapView: MLNMapView?
```

Comment:
Apple discourages keeping references to the UIView - the idea behind it is that the UIViewRepresentable should be written in a way so that iOS can discard the UIView at any time, and recreate it back to the state it needs to be in with calls to makeUIView and updateUIView at any time. So the UIView one might be keeping a reference to might no longer be the one that is actually on screen.

Sometimes it is impossible to adhere to this, and then one has to live with the risk (it rarely actually happens that iOS creates the UIView at "random"), but I don' think we need this workaround in this specific case? See my  `showTemporaryLocationAuthorization` State suggestion.

Reply:
I need to write a blog post about all this stuff I've found out with UIViewRepresentable and put my options there. For this example I have followed Apple's doc for now :) Appreciate the feedback.